### PR TITLE
🤯 Execution consts validation for update wg budget

### PR DESCRIPTION
**Pull request by WRadoslaw**
- #2327

### CODE REVIEW

File: packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx [hunk 1]
```diff
@@ -1,30 +1,49 @@
...
+  const currentBlock = useObservable(
+    () => api?.rpc.chain.subscribeNewHeads().pipe(map(({ number }) => number.toNumber())),
+    [api?.isConnected]
+  )
   const { group } = useWorkingGroup({ name: groupId })
 
+  const milisecondsLeft = useMemo(() => {
+    if (currentBlock && nextPaymentBlock) {
+      const blockToGo = nextPaymentBlock.toNumber() - currentBlock
+      return new Date(Date.now() + blockToGo * MILLISECONDS_PER_BLOCK).getTime() - Date.now()
```
Comment (thesan) [inline packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx:43]: Isn't this just?
```suggestion
      return (Number(nextPaymentBlock) - currentBlock) * MILLISECONDS_PER_BLOCK
```

File: packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx [hunk 2]
```diff
@@ -1,30 +1,49 @@
...
+  const nextPaymentBlock = useFirstObservableValue(() => api?.query.council.nextRewardPayments(), [api?.isConnected])
   const [groupId, isPositive, budgetUpdate] = watch([
     'updateWorkingGroupBudget.groupId',
     'updateWorkingGroupBudget.isPositive',
     'updateWorkingGroupBudget.budgetUpdate',
   ])
+  const currentBlock = useObservable(
+    () => api?.rpc.chain.subscribeNewHeads().pipe(map(({ number }) => number.toNumber())),
+    [api?.isConnected]
+  )
```
Comment (thesan) [inline packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx:37]:
```suggestion
  const currentBlock = useCurrentBlockNumber()
```

File: packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx [hunk 3]
```diff
@@ -81,25 +102,42 @@ export const UpdateWorkingGroupBudget = () => {
...
                 Current budget for Council is{' '}
                 <TextInlineMedium bold>
-                  <TokenValue value={councilBudget} />
+                  <TokenValue value={budget.amount} />
                 </TextInlineMedium>
                 .
               </TextMedium>
             </Info>
           )}
+          {nextPaymentBlock && milisecondsLeft && (
```
Comment (thesan) [inline packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx:124]: Just to make sure to not render "0":
```suggestion
          {nextPaymentBlock && !!milisecondsLeft && (
```

**Review by thesan — CHANGES_REQUESTED** [review]

File: packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx (hunk 1 above)
Comment (WRadoslaw) [reply packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx:43]: Yes, it is :)

**Review by thesan — APPROVED** [review]
:100:

**Pull request merged**